### ISSUE TITLE
sails-mongo fails to successfully locate a record the second time using findOne. If restarted, it finds it.

### ISSUE THREAD

**Issue by luislobo** [balderdashy/sails-mongo#125]
I've been having some erratic results using findOne.

You will find the code here:
https://github.com/luislobo/sails-mongo-error-findone

Test controller has an index action:
```
    index: function(req, res, next) {
        console.log("id:" + req.body.id);
        Test.findOne({id: req.body.id}, function foundTest(err, test) {
                if(!test) {
                    console.log('Test not found. CREATE IT!');
                    Test.create(req.body).exec(function testCreated(err, newTest){
                        if(err) {
                            console.log(err);
                        }
                        else if(newTest){

                            console.log("add success");
                        }
                        else {
                            console.log("could not add");
                        }
                    });
                } else {
                    console.log('Test found. UPDATE IT!');
                    Test.update(test.id, req.body).exec(function testUpdated(err, updatedTest){
                        if(err) {
                            console.log(err);
                        }
                        else if(updatedTest){
                            console.log("update success");
                        } else {
                            console.log("could not update!");
                        }
                    });
                }
                
        });
        res.send('DONE');
    }
```

These are my tests:

```curl -d "id=10&value=10" http://localhost:1337/test```

Console prints
```
id:10
Test not found. CREATE IT!
add success
```

MongoDB gets this document saved:
```
{
    value: "10",
    createdAt: ISODate("2014-04-01T20:30:53.626Z"),
    updatedAt: ISODate("2014-04-01T20:30:53.626Z"),
    _id: 10
}
```

Running again the same:

```curl -d "id=10&value=10" http://localhost:1337/test```

Console prints:
```
id:10
Test not found. CREATE IT!
{ originalError: 
   { [MongoError: E11000 duplicate key error index: myapp.test.$_id_  dup key: { : 10 }]
     name: 'MongoError',
     err: 'E11000 duplicate key error index: myapp.test.$_id_  dup key: { : 10 }',
     code: 11000,
     n: 0,
     connectionId: 250,
     ok: 1 },
  status: 'error',
  code: 'E_UNKNOWN',
  msg: 'Waterline: ORM encountered an unknown error.' }
```

Now, stop sails and start it again

Issue the same command
```curl -d "id=10&value=10" http://localhost:1337/test```

Console prints
```
id:10
Test found. UPDATE IT!
update success
```

MongoDB shows:

```
{
    value: "10",
    createdAt: ISODate("2014-04-01T20:30:53.626Z"),
    updatedAt: ISODate("2014-04-01T20:32:52.616Z"),
    _id: "10"
} 
```

I notice now that _id has a value of STRING 10.

The same code works always using sails-disk.

I've tested it with sails 0.9.7, 0.10.0-rc4 and 0.10.0-rc5, and latest sails-mongo 0.10.0-rc2. All with the same results.

**Comment by mikermcneil** [comment]
Thanks for posting, @luislobo.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails-mongo/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
